### PR TITLE
Replace remaining atomics in tasking with seqcst atomics

### PR DESCRIPTION
**Pull request by crtrott**
There were still random segfaults in some tasking tests
on power and arm.

### CODE REVIEW

File: core/src/impl/Kokkos_TaskQueueCommon.hpp [hunk 1]
```diff
@@ -87,6 +87,7 @@ class TaskQueueCommonMixin {
   // <editor-fold desc="Constructors, destructor, and assignment"> {{{2
 
   TaskQueueCommonMixin() : m_ready_count(0) {
+    Kokkos::memory_fence();
```
Comment (dalg24) [inline core/src/impl/Kokkos_TaskQueueCommon.hpp:90]: Did you mean to add this memory fence?

File: core/src/impl/Kokkos_TaskNode.hpp [hunk 2]
```diff
@@ -151,14 +151,19 @@ class ReferenceCountedBase {
   bool decrement_and_check_reference_count() {
     // TODO @tasking @memory_order DSH memory order
     auto old_count = Kokkos::atomic_fetch_add(&m_ref_count, -1);
+    Kokkos::memory_fence();
```
Comment (dalg24) [inline core/src/impl/Kokkos_TaskNode.hpp:154]: Why memory fence rather than a desul atomic?

File: core/src/impl/Kokkos_TaskQueueCommon.hpp (hunk 1 above)
Comment (crtrott) [reply core/src/impl/Kokkos_TaskQueueCommon.hpp:90]: yes

File: core/src/impl/Kokkos_TaskNode.hpp (hunk 2 above)
Comment (crtrott) [reply core/src/impl/Kokkos_TaskNode.hpp:154]: I wanted to limit the number of wrapper atomics I implement.

File: core/src/impl/Kokkos_TaskQueueCommon.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_TaskQueueCommon.hpp:90]: I don't get why it is needed but not blocking

**Review by dalg24 — APPROVED** [review]
You should probably open an issue to keep track of the TODOs for relaxing atomics

**Comment by crtrott** [comment]
Will do.

**Comment by crtrott** [comment]
#4308

**Pull request merged**